src/Aspire.Hosting/ApplicationModel/ResourceLoggerService.cs

```diff
@@ -359,6 +331,11 @@ private event Action<InternalLogLine> OnNewLog
 
                 if (raiseSubscribersChanged)
                 {
+                    // Clear backlog immediately.
+                    // Avoids a race between message being subscription changed notification eventually clearing the
+                    // logs and someone else watching logs and getting the backlog + complete replay off all logs.
+                    ClearBacklog();
```

Comment:
Isn't there still a race here? This happens outside of the lock. So we could release the lock above, and immediately a new subscriber could get added.

Reply:
Added lock around unsubscribe.